tensorflow/java/src/main/java/org/tensorflow/Tensor.java

```diff
@@ -317,7 +433,15 @@ private void throwExceptionIfTypeIsIncompatible(Object o) {
     }
   }
 
-  private static native long allocate(int dtype, long[] shape);
+  private static ByteBuffer getBuffer(long nativeHandle) {
+    return buffer(nativeHandle).order(ByteOrder.nativeOrder());
+  }
+
+  private static native ByteBuffer buffer(long handle);
+
+  private static native long allocate(int dtype, long[] shape, long sizeInBytes);
```

Comment:
Or maybe just have a single:

```java
private static native long allocate(int dtype, long[] shape, long sizeInBytes);
```

Since it is a private method, we can make it require that `sizeInBytes` be consistent with `shape` and `dtype`. So we don't need two methods, just one?